pkg/notifications/notification_test.go

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: 2024 Greenbone AG <https://greenbone.net>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+package notifications
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"sync/atomic"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClient_CreateNotification(t *testing.T) {
+	tests := []struct {
+		name                   string
+		serverHttpResponseCode int
+		wantErr                bool
+	}{
+		{
+			name:                   "notification can be sent",
+			serverHttpResponseCode: http.StatusOK,
+			wantErr:                false,
```

Comment:
personally I like it if either the error type is checked (see also comment concerning error handling above) - or the error message (regexp, or prefix check would be sufficient). Because in my mind only then you can know, that the right error occurred :-)

Reply:
If there is non-binary error handling required, I agrees that the error type or value (in case if a sentinel error) should be checked. However I'm against checking error messages itself, as an error message is more an implemenation detail.

So for now I would prefer to keep the check simple.